src/reyna/Connection/ConnectionInfo.cs

```diff
@@ -62,11 +68,83 @@ public bool Mobile
             }
         }
 
+        public bool Wifi
+        {
+            get
+            {
+                var networkInterfaces = NetworkInterface.GetAllNetworkInterfaces();
+                var wireless = ConnectionInfo.FindWireless(networkInterfaces);
+                if (wireless != null)
+                {
+                    return true;
+                }
+
+                var networkInterface = FindWirelessByExclusion(networkInterfaces);
+                if (networkInterface != null)
+                {
+                    return true;
+                }
+
+                return false;
+            }
+        }
+
+        public bool Roaming
+        {
+            get
+            {
+                var phoneRoamingBitMask = 0x200;
+                var status = this.Registry.GetDWord(Microsoft.Win32.Registry.LocalMachine, "System\\State\\Phone", "Status", 0);
+                return (status & phoneRoamingBitMask) == phoneRoamingBitMask;
+            }
+        }
+
+        internal IRegistry Registry { get; set; }
+
+        private static WirelessNetworkInterface FindWireless(INetworkInterface[] interfaces)
+        {
+            foreach (var ni in interfaces)
+            {
+                var wni = ni as WirelessNetworkInterface;
+                if (wni != null)
+                {
+                    return wni;
+                }
+            }
+
+            return null;
+        }
+
+        private static INetworkInterface FindWirelessByExclusion(INetworkInterface[] interfaces)
+        {
+            foreach (var ni in interfaces)
+            {
+                if (LANNetwork(ni) || ActiveSyncNetwork(ni) || GPRSNetwork(ni))
+                {
+                    continue;
+                }
+
+                return ni;
+            }
+
+            return null;
+        }
+
         private static bool GPRSNetwork(INetworkInterface ni)
         {
             return ni.Name.ToLower(System.Globalization.CultureInfo.CurrentCulture).StartsWith("cellular line", StringComparison.CurrentCulture);
         }
 
+        private static bool ActiveSyncNetwork(INetworkInterface ni)
+        {
+            return ni.Name.ToLower(System.Globalization.CultureInfo.CurrentCulture).StartsWith("usb", StringComparison.CurrentCulture);
+        }
+
+        private static bool LANNetwork(INetworkInterface ni)
```

Comment:
Seriously, we are detecting something as being a LAN network as being of a certain speed? Looks like those numbers are 10 and 100 MBit speeds (or at least 10 and 100 M of some unit, whatever that is). What about Gbit networks? Isn't that something fairly common now?